app/src/main/java/com/infomaniak/mail/utils/MessageBodyUtils.kt

```diff
@@ -91,6 +98,66 @@ object MessageBodyUtils {
         return htmlDocument.outerHtml() to quotes
     }
 
+    fun addPrintHeader(context: Context, message: Message, htmlDocument: Document) {
+        htmlDocument.body()
+            .attr("style", "margin: 40px")
+        htmlDocument.body().insertChildren(0, createPrintHeader(context, message))
+    }
+
+    private fun createPrintHeader(context: Context, message: Message): Element {
+        val rootHeaderDiv = Element("div")
+        val firstSeparator = Element("hr").attr("color", "black")
+        val secondSeparator = Element("hr").attr("color", "LightGray")
+
+        val iconElement = Element("img")
+            .attr("src", "file:///android_asset/icon_print_email.svg")
+            .attr("width", "150")
+        rootHeaderDiv.insertChildren(0, iconElement)
+        rootHeaderDiv.insertChildren(1, firstSeparator)
+
+        message.subject?.let { subject ->
+            val subjectElement = Element("b").appendText(subject)
+            rootHeaderDiv.insertChildren(2, subjectElement)
+        }
+
+        rootHeaderDiv.insertChildren(3, secondSeparator)
+
+        val messageDetailsDiv = Element("div").attr("style", "margin-bottom: 40px; display: block")
+        messageDetailsDiv.insertPrintRecipientField(context.getString(R.string.ccTitle), *message.cc.toTypedArray())
+        messageDetailsDiv.insertPrintRecipientField(context.getString(R.string.toTitle), * message.to.toTypedArray())
+        message.sender?.let { messageDetailsDiv.insertPrintRecipientField(context.getString(R.string.fromTitle), it) }
+
+        val formattedDate = message.date.toDate().formatNumericalDayMonthYearWithTime()
+        messageDetailsDiv.insertPrintDateField(context.getString(R.string.dateTitle), formattedDate)
+
+        rootHeaderDiv.insertChildren(4, messageDetailsDiv)
+
+        rootHeaderDiv.attr("style", "margin-bottom: 40px")
+
+        return rootHeaderDiv
+    }
+
+    private fun Element.insertPrintRecipientField(prefix: String, vararg recipients: Recipient) {
+        if (recipients.isEmpty()) return
+
+        insertChildren(
+            0,
+            insertField(prefix).appendText(
+                recipients.joinToString { recipient -> recipient.quotedDisplay() }
+            ),
+        )
+    }
+
+    private fun Element.insertPrintDateField(prefix: String, date: String) {
+        insertChildren(0, insertField(prefix).appendText(date))
+    }
```

Comment:
The same way `insertPrintRecipientField` takes recipients as input, shouldn't `insertPrintDateField` take a Date object as input?